### PR TITLE
python311Packages.datasets: 2.12.0 -> 2.14.4

### PR DESCRIPTION
**Pull request by fabaff**
Diff: https://github.com/huggingface/datasets/compare/refs/tags/2.12.0...2.14.4

Changelog: https://github.com/huggingface/datasets/releases/tag/2.14.4

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 252434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
  </ul>
</details>
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>openai-whisper.dist (python310Packages.openai-whisper.dist)</li>
    <li>python310Packages.accelerate</li>
    <li>python310Packages.accelerate.dist</li>
    <li>python310Packages.anthropic</li>
    <li>python310Packages.anthropic.dist</li>
    <li>python310Packages.datasets</li>
    <li>python310Packages.datasets.dist</li>
    <li>python310Packages.evaluate</li>
    <li>python310Packages.evaluate.dist</li>
    <li>python310Packages.manifest-ml</li>
    <li>python310Packages.manifest-ml.dist</li>
    <li>python310Packages.minichain</li>
    <li>python310Packages.minichain.dist</li>
    <li>python310Packages.optimum</li>
    <li>python310Packages.optimum.dist</li>
    <li>python310Packages.peft</li>
    <li>python310Packages.peft.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.stanza.dist</li>
    <li>python310Packages.tokenizers</li>
    <li>python310Packages.tokenizers.dist</li>
    <li>python310Packages.transformers</li>
    <li>python310Packages.transformers.dist</li>
    <li>python310Packages.txtai</li>
    <li>python310Packages.txtai.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python311Packages.anthropic</li>
    <li>python311Packages.anthropic.dist</li>
    <li>python311Packages.datasets</li>
    <li>python311Packages.datasets.dist</li>
    <li>python311Packages.evaluate</li>
    <li>python311Packages.evaluate.dist</li>
    <li>python311Packages.optimum</li>
    <li>python311Packages.optimum.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.tokenizers</li>
    <li>python311Packages.tokenizers.dist</li>
    <li>python311Packages.transformers</li>
    <li>python311Packages.transformers.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 252434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>64 packages marked as broken and skipped:</summary>
  <ul>
    <li>libretranslate</li>
    <li>libretranslate.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python310Packages.argos-translate-files</li>
    <li>python310Packages.argos-translate-files.dist</li>
    <li>python310Packages.argostranslate</li>
    <li>python310Packages.argostranslate.dist</li>
    <li>python310Packages.cleanlab</li>
    <li>python310Packages.cleanlab.dist</li>
    <li>python310Packages.ctranslate2</li>
    <li>python310Packages.ctranslate2.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.faster-whisper</li>
    <li>python310Packages.faster-whisper.dist</li>
    <li>python310Packages.gradio</li>
    <li>python310Packages.gradio.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.libretranslate</li>
    <li>python310Packages.libretranslate.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.translatehtml</li>
    <li>python310Packages.translatehtml.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.gradio</li>
    <li>python311Packages.gradio.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>openai-whisper.dist (python310Packages.openai-whisper.dist)</li>
    <li>python310Packages.accelerate</li>
    <li>python310Packages.accelerate.dist</li>
    <li>python310Packages.anthropic</li>
    <li>python310Packages.anthropic.dist</li>
    <li>python310Packages.datasets</li>
    <li>python310Packages.datasets.dist</li>
    <li>python310Packages.evaluate</li>
    <li>python310Packages.evaluate.dist</li>
    <li>python310Packages.manifest-ml</li>
    <li>python310Packages.manifest-ml.dist</li>
    <li>python310Packages.minichain</li>
    <li>python310Packages.minichain.dist</li>
    <li>python310Packages.optimum</li>
    <li>python310Packages.optimum.dist</li>
    <li>python310Packages.peft</li>
    <li>python310Packages.peft.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.stanza.dist</li>
    <li>python310Packages.tokenizers</li>
    <li>python310Packages.tokenizers.dist</li>
    <li>python310Packages.transformers</li>
    <li>python310Packages.transformers.dist</li>
    <li>python310Packages.txtai</li>
    <li>python310Packages.txtai.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.anthropic</li>
    <li>python311Packages.anthropic.dist</li>
    <li>python311Packages.datasets</li>
    <li>python311Packages.datasets.dist</li>
    <li>python311Packages.evaluate</li>
    <li>python311Packages.evaluate.dist</li>
    <li>python311Packages.optimum</li>
    <li>python311Packages.optimum.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.tokenizers</li>
    <li>python311Packages.tokenizers.dist</li>
    <li>python311Packages.transformers</li>
    <li>python311Packages.transformers.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/datasets/default.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 buildPythonPackage rec {
   pname = "datasets";
-  version = "2.12.0";
+  version = "2.14.4";
   format = "setuptools";
 
   disabled = pythonOlder "3.7";
```
Comment (natsukium) [inline pkgs/development/python-modules/datasets/default.nix:27]:
```suggestion
  disabled = pythonOlder "3.8";
```
https://github.com/huggingface/datasets/blob/53d55f33bfac9febb0c355e136f2847e5f3e3b53/setup.py#L265

Besides this, LGTM.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 252434` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>61 packages built:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>openai-whisper.dist (python310Packages.openai-whisper.dist)</li>
    <li>python310Packages.accelerate</li>
    <li>python310Packages.accelerate.dist</li>
    <li>python310Packages.anthropic</li>
    <li>python310Packages.anthropic.dist</li>
    <li>python310Packages.datasets</li>
    <li>python310Packages.datasets.dist</li>
    <li>python310Packages.evaluate</li>
    <li>python310Packages.evaluate.dist</li>
    <li>python310Packages.manifest-ml</li>
    <li>python310Packages.manifest-ml.dist</li>
    <li>python310Packages.minichain</li>
    <li>python310Packages.minichain.dist</li>
    <li>python310Packages.optimum</li>
    <li>python310Packages.optimum.dist</li>
    <li>python310Packages.peft</li>
    <li>python310Packages.peft.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.stanza.dist</li>
    <li>python310Packages.tokenizers</li>
    <li>python310Packages.tokenizers.dist</li>
    <li>python310Packages.transformers</li>
    <li>python310Packages.transformers.dist</li>
    <li>python310Packages.txtai</li>
    <li>python310Packages.txtai.dist</li>
    <li>python310Packages.unstructured-inference</li>
    <li>python310Packages.unstructured-inference.dist</li>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.anthropic</li>
    <li>python311Packages.anthropic.dist</li>
    <li>python311Packages.datasets</li>
    <li>python311Packages.datasets.dist</li>
    <li>python311Packages.evaluate</li>
    <li>python311Packages.evaluate.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.optimum</li>
    <li>python311Packages.optimum.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.tokenizers</li>
    <li>python311Packages.tokenizers.dist</li>
    <li>python311Packages.transformers</li>
    <li>python311Packages.transformers.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python311Packages.unstructured-inference</li>
    <li>python311Packages.unstructured-inference.dist</li>
    <li>unstructured-api</li>
  </ul>
</details>

**Pull request merged**